Sources/Dependency/source/Analysis.cs

```diff
@@ -351,7 +351,8 @@ public static List<Tuple<string,string, int>> PopulateChangeLog(string changelis
                     string procName = items[0].Trim(), procFile = null;
                     // locate the source file for the procedure
                     var impl = program.Implementations.Single(i => i.Name == procName);
-                    impl.Blocks.Find(b => b.Cmds.Count > 0 && b.Cmds[0] is AssertCmd && (procFile = Utils.AttributeUtils.GetSourceFile(b)) != null);
+                    // TODO: the following doesn't do anything?
+                    //impl.Blocks.Find(b => b.Cmds.Count > 0 && b.Cmds[0] is AssertCmd && (procFile = Utils.AttributeUtils.GetSourceFile(b)) != null);
```

Comment:
The condition `(procFile = Utils.AttributeUtils.GetSourceFile(b)) != null` is scary. Isn't it assigning to `procFile`?

Reply:
Oh, you're right! Good catch.